src/services/formatting/formatting.ts

```diff
@@ -438,6 +438,25 @@ namespace ts.formatting {
             }
         }
 
+        if (previousRange! && formattingScanner.getStartPos() >= originalRange.end) {
+            const token =
+                formattingScanner.isOnEOF() ? formattingScanner.readEOFTokenRange() :
+                formattingScanner.isOnToken() ? formattingScanner.readTokenInfo(enclosingNode).token :
+                undefined;

```

Comment:
Here’s where that block of code moved to, with some minor changes. We previously handled edits at the end of a _file_ in this block by getting a token representation of the EOF token and passing it to `processPair`. But for the same to work with formatting a _selection_, we’ll probably be dealing with a regular token instead of an EOF.